src/Utilities/ResourceCollection.h

```diff
@@ -15,6 +15,7 @@
 #include <string>
 #include <memory>
 #include <cstddef>
+#include <cassert>
```

Comment:
Is this the correct place? Don't feel right adding such include just for convenience.

Reply:
It was the wrong place.  Fixed now.